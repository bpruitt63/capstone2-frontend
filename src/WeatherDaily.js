import React, {useState} from 'react';
import {Card, CardText, CardBody, CardTitle, Row, Col, Button, Collapse} from 'reactstrap';
import './static/styles/Weather.css';
import {makeReadableDate} from './static/helpers';

/** Component for a single day's weather forecast */
function WeatherDaily({daily, units}) {

    const [isOpen, setIsOpen] = useState(false);

    /** Toggles collapsible weather */
    const toggle = () => setIsOpen(!isOpen);

    const ICON_URL = 'https://www.weatherbit.io/static/img/icons/'
    const date = makeReadableDate(daily.valid_date);

    return (
        <Card>
            {!isOpen &&
                <CardBody className='closedCard'>
                    <CardTitle tag='h5'>{date}</CardTitle>
                    <Row>
                        <Col xs={3}>
                            <img src={`${ICON_URL}${daily.weather.icon}.png`} 
                                alt={daily.weather.description}
                                className='closedImg' />
                        </Col>
                        <Col xs={2} md={3}>
                            <CardText>High
                                <span><br/>{daily.high_temp}</span>
                            </CardText>
                        </Col>
                        <Col xs={2} md={3}>
                            <CardText>Wind
                                <span><br/>{daily.wind_spd}
                                    {units === 'i' ? 'mph ' : 'km/h '}
                                    {daily.wind_cdir}
                                </span>
                            </CardText>
                        </Col>
                        <Col xs={5} md={3}>
                            <Button className='dropdown-toggle WeatherToggle' onClick={toggle}>
                                See More
                            </Button>
                        </Col>
                    </Row>
                </CardBody>
            }
            <Collapse isOpen={isOpen}>
                <CardBody className='openCard'>
                    <CardTitle tag='h5'>{date}</CardTitle>
                    <img src={`${ICON_URL}${daily.weather.icon}.png`} 
                                alt={daily.weather.description}
                                className='openImg' />
                    <CardText>
                        {daily.weather.description}
                    </CardText>
                    <CardText>
                        Precipitation chance: {daily.pop}%
                    </CardText>
                    <CardText>
                        High: {daily.high_temp}, feels like {daily.app_max_temp}
                    </CardText>
                    <CardText>
                        Low: {daily.low_temp}, feels like {daily.app_min_temp}
                    </CardText>
                    <CardText>
                        Wind: {daily.wind_spd}
                        {units === 'i' ? 'mph ' : 'km/h '}{daily.wind_cdir}
                        , gust speed {daily.wind_gust_spd}{units === 'i' ? 'mph ' : 'km/h '}
                    </CardText>
                    <Button className='dropdown-toggle WeatherToggle' onClick={toggle}>
                        See Less
                    </Button>
                </CardBody>
            </Collapse>     
        </Card>
    );
};

export default WeatherDaily;