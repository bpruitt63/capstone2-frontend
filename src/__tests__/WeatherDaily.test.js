import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import WeatherDaily from '../WeatherDaily';

const daily = {wind_spd: 10, wind_gust_spd: 15, wind_cdir: 'N',
                        high_temp: 80, app_max_temp: 90, low_temp: 60, 
                        app_min_temp: 55, pop: 15, valid_date: '2012-01-01',
                        weather: {
                            icon: 'none',
                            description: 'test description'
                        }};

test('renders', () => {
    render(<WeatherDaily daily={daily}/>);
});

test('matches snapshot', () => {
    const {asFragment} = render(<WeatherDaily daily={daily}/>);
    expect(asFragment()).toMatchSnapshot();
});

test('renders w/ correct date', () => {
    render(<WeatherDaily daily={daily}/>);
    expect(screen.getAllByText(/January 01, 2012/)).toBeInTheDocument;
});

test('toggles', () => {
    render(<WeatherDaily daily={daily}/>);
    expect(screen.getAllByText(/80/)).toBeInTheDocument;
    expect(screen.queryByText(/90/)).not.toBeInTheDocument;
    expect(screen.getByText(/See More/)).toBeInTheDocument;
    const toggleButton = screen.getByText(/See More/);
    fireEvent.click(toggleButton);
    expect(screen.getByText(/80/)).toBeInTheDocument;
    expect(screen.getByText(/90/)).toBeInTheDocument;
    expect(screen.queryByText(/See More/)).not.toBeInTheDocument;
    expect(screen.getByText(/See Less/)).toBeInTheDocument;
});