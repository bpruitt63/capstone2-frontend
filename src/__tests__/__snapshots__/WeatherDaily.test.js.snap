// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="closedCard card-body"
    >
      <h5
        class="card-title"
      >
        January 01, 2012
      </h5>
      <div
        class="row"
      >
        <div
          class="col-3"
        >
          <img
            alt="test description"
            class="closedImg"
            src="https://www.weatherbit.io/static/img/icons/none.png"
          />
        </div>
        <div
          class="col-2 col-md-3"
        >
          <p
            class="card-text"
          >
            High
            <span>
              <br />
              80
            </span>
          </p>
        </div>
        <div
          class="col-2 col-md-3"
        >
          <p
            class="card-text"
          >
            Wind
            <span>
              <br />
              10km/h N
            </span>
          </p>
        </div>
        <div
          class="col-5 col-md-3"
        >
          <button
            class="dropdown-toggle WeatherToggle btn btn-secondary"
            type="button"
          >
            See More
          </button>
        </div>
      </div>
    </div>
    <div
      class="collapse"
    >
      <div
        class="openCard card-body"
      >
        <h5
          class="card-title"
        >
          January 01, 2012
        </h5>
        <img
          alt="test description"
          class="openImg"
          src="https://www.weatherbit.io/static/img/icons/none.png"
        />
        <p
          class="card-text"
        >
          test description
        </p>
        <p
          class="card-text"
        >
          Precipitation chance: 15%
        </p>
        <p
          class="card-text"
        >
          High: 80, feels like 90
        </p>
        <p
          class="card-text"
        >
          Low: 60, feels like 55
        </p>
        <p
          class="card-text"
        >
          Wind: 10km/h N, gust speed 15km/h 
        </p>
        <button
          class="dropdown-toggle WeatherToggle btn btn-secondary"
          type="button"
        >
          See Less
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
